src/utils/sheets.js

```diff
@@ -0,0 +1,47 @@
+import c from '../config'
+import {google} from 'googleapis'
+
+// Authentication
+// Google Sheet must give read permission to the service account in c.google.email
+const scopes = ['https://www.googleapis.com/auth/spreadsheets.readonly']
+const key = Buffer.from(c.google.key, 'base64').toString()
+const auth = new google.auth.JWT(c.google.email, null, key, scopes)
+const sheetsApi = google.sheets({version: 'v4', auth})
+
+/**
+ * Loads data from Google Sheets
+ * Returns an array of objects (rows) indexed by column headers (first row) 
+ * Called from contract template
+ * @param {*} people
+ * @param {spreadsheetId, sheetId, range, idSheetsColumn, idEMSField} sheet: Google sheet and range identifier, fields to join with EMS
+ */
+export const loadSheetData = async (people, sheet) => {
+  const personIDs = people.map((person) => person[sheet.idEMSField])
+
+  let [header, ...values] = await getValues(sheet.spreadsheetId, sheet.range)
+
+  // Index by column headers,
+  // filter rows in the array people - from the original request
+  values = values
+    .map((row) => {
+      let obj = {}
+      header.forEach((key, col) => {
+        obj[key] = row[col]
+      })
+      return obj
+    })
```

Comment:
This looks like a good place to use an [Array.reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce). (I'm pretty sure this should work, but haven't checked.)
```suggestion
    .map((row) =>
      header.reduce((obj, key, col) => {
        obj[key] = row[col]
      }, {})
    )
```

Reply:
that's smart, almost works